quickwit-cluster/src/cluster.rs

```diff
@@ -206,6 +223,22 @@ impl Cluster {
                 .node_state(&NodeId::from(member.clone()))
                 .unwrap();
 
+            let available_services = node_state
+                .get(AVAILABLE_SERVICES_KEY)
+                .ok_or_else(|| {
```

Comment:
In which cases is this key missing? This would be a bug, right?

Reply:
Yes, it has to be there

Reply:
It is not necessarily a bug.
If a user just joined, we might have been sending partial updates.
In that case, the member might be there, but we might not know its services.

We can try to frown really hard and try to find reason for it to never happen, but I think the right mental model here is that it might happen.

Returning an error here is not a good a idea.

Reply:
There's also the case of rolling updates. During a rolling update, version X and version Y need to coexist. So if version Y adds a new service (maybe metastore or coordinator), version X will throw some errors when it probably should ignore the new service.

Reply:
Yes or an adversarial payload, a faulty dev server, etc. 
I think the rule of thumb is that whenever possible, we need to treat anything stuff coming from the network very cautiously.

In this case, a simple way to handle this is to assume that if the key is not there yet then we assume no service is exposed yet.